service/tcp.go

```diff
@@ -57,10 +57,23 @@ func debugTCP(cipherID, template string, val interface{}) {
 	}
 }
 
+func calculateTCPTrialSize(ciphers []*list.Element) int {
```

Comment:
I think this can be computed statically from `ss.supportedAEADs`.  I would suggest making the trial size a fixed global calculated on startup.

Reply:
This function has minimal impact on the cpu usage, so I'm inclined to keep it. This way we read less bytes if the user is using a smaller cipher.

Reply:
I don't think it helps to read fewer bytes.  We're going to have to read those bytes eventually, and the required-vs-provided test proves that they have to be sent.

For Outline, we always use the maximum-size cipher, so the output of this function will always be 50.  Even if the CPU impact is small, I think it's bad for readability to do an O(N) operation on every connection to compute a value that is actually a compile-time constant.

Reply:
Done